src/client.rs

```diff
@@ -1097,22 +1124,35 @@ impl Client {
                     // NOTE: we must not authenticate on additional URLs as those
                     // can be abused to leak credentials or tokens.  Please
                     // refer to CVE-2020-15157 for more information.
-                    response =
+                    request =
                         RequestBuilderWrapper::from_client(self, |client| client.get(url.clone()))
                             .apply_accept(MIME_TYPES_DISTRIBUTION_MANIFEST)?
-                            .into_request_builder()
-                            .send()
-                            .await?
+                            .into_request_builder();
+                    if let Some(offset) = offset {
+                        request = request.header(
+                            RANGE,
+                            HeaderValue::from_str(&format!("bytes={offset}-")).unwrap(),
+                        );
+                    }
+                    response = request.send().await?
                 }
             }
         }
 
+        Ok(response)
+    }
+
+    /// Converts a response into a stream
+    fn stream_from_response(&self, response: Response) -> Result<SizedStream> {
```

Comment:
Nit: This can probably be a helper function that lives outside of the `Client` impl since it doesn't depend on anything in `self`

Reply:
Yeah, that are probably old habits from other programming languages, like Java for example, where just everything is in the scope of a class, so I just didn't think about that I could move this helper function outside and remove `&self`, so thanks for noticing :smile: 

(the only advantage it had with being inside of `Client` was, that it was close to where it was used, so it's maybe easier to read the code when you don't need to jump around that much, but with modern IDEs that's probably not a big problem)